service/history/circuitbreakerpool/fx.go

```diff
@@ -0,0 +1,74 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package circuitbreakerpool
+
+import (
+	"fmt"
+
+	"go.temporal.io/server/common/circuitbreaker"
+	"go.temporal.io/server/common/namespace"
+	"go.temporal.io/server/service/history/configs"
+	"go.temporal.io/server/service/history/tasks"
+	"go.uber.org/fx"
+)
+
+var Module = fx.Options(
+	fx.Provide(OutboundQueueCircuitBreakerPoolProvider),
+)
+
+type OutboundQueueCircuitBreakerPool struct {
```

Comment:
I think this can just be a type alias but not critical.

Reply:
So, if it is a type alias, there will be an issue in fx dependency injection if someone wants to create another circuit breaker with the same key, because the roots would be effectively the same.

Reply:
Looks like it's working here: https://github.com/temporalio/temporal/blob/5561bd78af81221d1c566b8e614632c84c3d9719/common/resource/fx.go#L82-L83

Reply:
This works because it's not an alias, it's still a type definition to an interface, which makes work nicer because it won't need to implement the methods. Im my case, I'm type def'ing to a struct, so it's just easier to embed so it will inherit the methods.